aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -93,146 +165,126 @@ module AptosFramework::Token {
+        creator: address,
+        collection_name: String,
+        uri: String,
+        description: String,
+        maximum: u64
     }
 
     /// Capability required to mint tokens.
     struct MintCapability has store {
-        token_id: TokenId,
+        token_id: TokenId
     }
 
     /// Capability required to burn tokens.
     struct BurnCapability has store {
-        token_id: TokenId,
+        token_id: TokenId
     }
 
     //
     // Creator Script functions
     //
 
-    public(script) fun create_limited_collection_script(
+    public(script) fun create_collection_script(
         creator: &signer,
         name: vector<u8>,
         description: vector<u8>,
         uri: vector<u8>,
         maximum: u64,
+        mutate_setting: vector<bool>
     ) acquires Collections {
         create_collection(
             creator,
             ASCII::string(name),
             ASCII::string(description),
             ASCII::string(uri),
-            Option::some(maximum),
-        );
-    }
-
-    public(script) fun create_unlimited_collection_script(
-        creator: &signer,
-        name: vector<u8>,
-        description: vector<u8>,
-        uri: vector<u8>,
-    ) acquires Collections {
-        create_collection(
-            creator,
-            ASCII::string(name),
-            ASCII::string(description),
-            ASCII::string(uri),
-            Option::none(),
+            maximum,
+            mutate_setting
         );
     }
 
-    public(script) fun create_limited_token_script(
+    public(script) fun create_token_script(
         creator: &signer,
         collection: vector<u8>,
         name: vector<u8>,
         description: vector<u8>,
-        monitor_supply: bool,
-        initial_balance: u64,
+        balance: u64,
         maximum: u64,
         uri: vector<u8>,
-        royalty_points_per_million: u64
+        royalty_payee_address: address,
+        royalty_points_denominator: u64,
+        royalty_points_nominator: u64,
+        token_mutate_setting: vector<bool>,
+        property_keys: vector<String>,
+        property_values: vector<vector<u8>>,
+        property_types: vector<String>
     ) acquires Collections, TokenStore {
         create_token(
             creator,
             ASCII::string(collection),
             ASCII::string(name),
             ASCII::string(description),
-            monitor_supply,
-            initial_balance,
-            Option::some(maximum),
+            balance,
+            maximum,
             ASCII::string(uri),
-            royalty_points_per_million
+            royalty_payee_address,
+            royalty_points_denominator,
+            royalty_points_nominator,
+            token_mutate_setting,
+            property_keys,
+            property_values,
+            property_types,
         );
     }
 
-    public(script) fun create_unlimited_token_script(
-        creator: &signer,
+    public(script) fun mint(
```

Comment:
I think we need to have
* create -> new token / tokenid / tokendata
* mint -> add more tokens to tokenid
* mint_instance -> add new serial number

Reply:
to be determined for the serial_number implementation